src/redux/reducers/siloDomainData/entries.js

```diff
@@ -261,6 +269,63 @@ const deleteEntry = (state, action) => {
     return newState;
 };
 
+const editEntry = (state, action) => {
+    const {
+        entry,
+        entryId,
+        leadId,
+    } = action;
+
+    const { activeProject: projectId } = state;
+    const {
+        entriesView: {
+            [projectId]: {
+                entries: leadGroupedEntries = [],
+            } = {},
+        } = {},
+    } = state;
+
+    const newState = produce(state, (safeState) => {
+        if (!safeState.entriesView) {
+            // eslint-disable-next-line no-param-reassign
+            safeState.entriesView = {
+            };
+        }
+        if (!safeState.entriesView[projectId]) {
+            // eslint-disable-next-line no-param-reassign
+            safeState.entriesView[projectId] = {
+                totalEntriesCount: 0,
+            };
+        }
+        if (!safeState.entriesView[projectId].entries) {
+            // eslint-disable-next-line no-param-reassign
+            safeState.entriesView[projectId].entries = [];
+        }
+        const safeLeads = safeState.entriesView[projectId].entries;
+        const leadIndex = leadGroupedEntries.findIndex(l => leadId === l.id);
+        if (leadIndex > -1) {
+            if (!safeLeads[leadIndex].entries) {
+                // eslint-disable-next-line no-param-reassign
+                safeLeads[leadIndex].entries = [];
+            }
+
+            const entryIndex = safeLeads[leadIndex].entries.findIndex(e => entryId === e.id);
+            const safeEntries = safeLeads[leadIndex].entries;
+
+            if (entryIndex > -1) {
+                // eslint-disable-next-line no-param-reassign
+                safeEntries.splice(entryIndex, 1);
+                safeEntries[entryIndex] = {
+                    ...entry,
+                    lead: safeEntries[entryIndex].lead,
+                };
```

Comment:
What is the intention here?

Reply:
The lead data is not changed while editing, but we fetch leads with a bit more detail.
So, we are just merging the edited entry data with old lead data.